pkg/server/user/user.go

```diff
@@ -61,23 +61,32 @@ func (dbu *DatabaseOperations) GetUser(email string) (*database.User, error) {
 	return u, nil
 }
 
-func (dbu *DatabaseOperations) SetPassword(email, newPassword string) error {
+func (dbu *DatabaseOperations) SetPassword(user *database.User, newPassword, userTarget string) error {
+	email := user.Email
+	if userTarget != "" && userTarget != email {
+		if !user.IsAdmin {
+			return auth.ErrPermissionDenied
+		}
+		email = userTarget
+	}
+
 	u, err := dbu.GetUser(email)
 	if err != nil {
 		return err
 	}
+
 	pass, err := bcrypt.GenerateFromPassword([]byte(newPassword), bcrypt.DefaultCost)
 	if err != nil {
 		return teresa_errors.New(
 			teresa_errors.ErrInternalServerError,
-			errors.Wrap(err, fmt.Sprintf("Generating the password hash to user %s", email)),
+			errors.Wrap(err, fmt.Sprintf("Generating the password hash to user %s", u.Email)),
```

Comment:
I guess it's email on errors.Wrap as email may be different from u.Email, right?

Reply:
Yes, I'll fix this, thanks!